AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -1335,17 +1381,21 @@ - (ASCellNodeBlock)dataController:(ASDataController *)dataController nodeBlockAt
   // Wrap the node block
   __weak __typeof__(self) weakSelf = self;
   return ^{
-    __typeof__(self) strongSelf = weakSelf;
     ASCellNode *node = (block != nil ? block() : [[ASCellNode alloc] init]);
-    [node enterHierarchyState:ASHierarchyStateRangeManaged];
-    if (node.interactionDelegate == nil) {
-      node.interactionDelegate = strongSelf;
-    }
+    [weakSelf didCreateNode:node];
     return node;
   };
   return block;
 }
 
+- (void)didCreateNode:(ASCellNode *)node
+{
+  [node enterHierarchyState:ASHierarchyStateRangeManaged];
+  if (node.interactionDelegate == nil) {
```

Comment:
Is this check important?  Should it check != self?  I wonder what happens if people are reusing ASCellNodes across different data controllers - this would be bad, but ideally either explicitly / immediately bad (or works!), rather than implicitly / subtly bad :)